third_party/move/move-prover/bytecode-pipeline/src/number_operation_analysis.rs

```diff
@@ -461,15 +461,41 @@ impl<'a> NumberOperationAnalysis<'a> {
                                 for (arg, arg_oper) in args.iter().zip(arg_oper.iter()) {
                                     if merged != *arg_oper {
                                         // propagate to arg if necessary
-                                        global_state.update_node_oper(
-                                            arg.node_id(),
-                                            merged,
-                                            allow_merge,
+                                        let mut update_flag = matches!(
+                                            arg.clone().into(),
+                                            ExpData::Temporary(..)
+                                                | ExpData::LocalVar(..)
+                                                | ExpData::Value(..)
+                                                | ExpData::Call(_, ASTOperation::Cast, _)
                                         );
-                                        update_temporary(arg, &merged, global_state, state);
+                                        if let ExpData::Call(
+                                            _,
+                                            ASTOperation::SpecFunction(mid, sid, _),
+                                            _,
+                                        ) = arg.clone().into()
+                                        {
```

Comment:
This is a pretty convoluted expression.  Perhaps a helper function with an informative name would be useful?  Or at least a comment to say what you are checking here.

Reply:
done